### ISSUE TITLE
Next execution date skips a year if configured for January 1st at 00:00, but only from December 1st at 00:01

### ISSUE THREAD

**Issue by MikyKuroneko** [poliander/cron#14]
I've run into a strange problem while calculating the next execution date. I have a cron configured for January 1st at 00:00. It was working well before, saying that next execution would be on 2023-01-01 00:00. Today, however, it says 2024-01-01 00:00.  
  
Using the existing test for the class I was able to pinpoint the breaking point. The calculation for next year works well until December 1st at 00:00:59. Past this time, the class will calculate the next execution skipping one year.  
  
To reproduce, you can add this to `getNextProvider` on `CronExpressionTest.php`:  
  
```php
['0 0 1 1 *', 1669849200, 1672527600], // on 2022-12-01 00:00:00, next execution at 2023-01-01 00:00:00
```  
  
This one works well. `1669849200` is December 1st, 2022 at 00:00. `1672527600` is January 1st, 2023 at 00:00, as expected.  
  
```php
['0 0 1 1 *', 1669849260, 1672527600], // on 2022-12-01 00:01:00, next execution at 2024-01-01 00:00:00
```  
  
This one fails. `1669849260` is one minute later, December 1st, 2022 at 00:01. This results in `1704063600`, which corresponds to January 1st, 2024 at 00:00.

**Comment by poliander** [comment]
Thanks for reporting, I will look into it.

**Comment by poliander** [comment]
@MikyKuroneko 

It should be fixed in 2.4.x-dev and dev-main, could you please take a look at it to confirm? I also included your test cases into the unit tests.

**Comment by poliander** [comment]
Fixed in 2.4.5 and 3.0.3

**Comment by MikyKuroneko** [comment]
It is indeed working correctly now. Thank you!
